prow/git/v2/interactor.go

```diff
@@ -54,6 +54,8 @@ type Interactor interface {
 	Fetch() error
 	// FetchRef fetches the refspec
 	FetchRef(refspec string) error
+	// FetchFromRemote fetches the branch of the given remote
+	FetchFromRemote(remote RemoteResolver, branch string) error
```

Comment:
Why isn't this remote part of the client when it's constructed?

Reply:
It could already have a different remote when it is constructed. So before this, you couldn't just have a cloned remote and fetch from different remote.

Reply:
One use-case is:

- Clone a repo from remote x
- Fetch a branch from remote y
- Merge
- Push branch to remote x